src/smtp/send.rs

```diff
@@ -44,14 +57,17 @@ impl Smtp {
             message,
         );
         if let Some(ref mut transport) = self.transport {
-            transport.send(mail).await.map_err(Error::SendError)?;
-
-            context.call_cb(Event::SmtpMessageSent(format!(
-                "Message len={} was smtp-sent to {}",
-                message_len, recipients_display
-            )));
+            let res = async_std::future::timeout(Duration::from_secs(SEND_TIMEOUT), async move {
+                transport.send(mail).await.map_err(Error::SendError)
+            })
+            .await?;
 
-            Ok(())
+            res.map(|_response| {
+                context.call_cb(Event::SmtpMessageSent(format!(
+                    "Message len={} was smtp-sent to {}",
+                    message_len, recipients_display
+                )));
+            })
```

Comment:
not sure that the error-handling and the "async move" is the best way to do this, but it seems to work.

Reply:
Fixed it.